sensor/kubernetes/sensor/sensor.go

```diff
@@ -198,12 +196,30 @@ func CreateSensor(cfg *CreateOptions) (*sensor.Sensor, error) {
 	} else {
 		networkFlowService = service.NewService(networkFlowManager, service.WithTraceWriter(cfg.networkFlowWriter))
 	}
+
+	var signalSrv signalService.Service
+	if cfg.signalServiceAuthFuncOverride != nil && cfg.localSensor {
+		signalSrv = signalService.NewService(signalCmp.GetReceiver(),
```

Comment:
Okay, so you create a channel in `signalCmp` only to pass it to another object so that it can write to it?
This is not a good idea, we must always make sure that:
- There is exactly one writer of the channel. That writer is responsible for creating the channel and closing the channel. Writer returns the channel to the outside always as read-only channel.
- There can be many readers of the channel and none of them should attempt to write to it or close it.
- Writer and readers must be easy to identify.

For more details check https://www.oreilly.com/library/view/concurrency-in-go/9781491941294/ch04.html

Reply:
As I explained offline, my initial approach was to have a single channel owned by the component and have it read messages coming from collector from both the old and new services. Thanks for taking the time to explain why this would be a dangerous practice in Go, I've made it so now the services each own a channel and those are passed in to the component for handling.